vortex-array/src/array/mod.rs

```diff
@@ -402,8 +402,9 @@ impl<V: VTable> Array for ArrayAdapter<V> {
                         stat,
                         Stat::IsConstant | Stat::IsSorted | Stat::IsStrictSorted
                     ) && value.as_ref().as_exact().is_some_and(|v| {
-                        v.as_bool()
-                            .vortex_expect("must be a bool")
+                        Scalar::new(DType::Bool(Nullability::NonNullable), v.clone())
```

Comment:
`Scalar::bool`

Reply:
This is one of the bad parts of this pr - the iterator yields you ScalarValues and I removed any way to unwrap a ScalarValue so you have to wrap it in a Scalar to unwrap